script/tool/lib/src/create_all_plugins_app_command.dart

```diff
@@ -155,7 +155,7 @@ class CreateAllPluginsAppCommand extends PluginCommand {
       version: Version.parse('1.0.0+1'),
       environment: <String, VersionConstraint>{
         'sdk': VersionConstraint.compatibleWith(
-          Version.parse('2.12.0'),
+          Version.parse('2.17.0'),
```

Comment:
I think what we'll want to do is read `flutter --version --machine` and then conditionally set this to 12 or 17 depending on the `dartSDKVersion` we get.

Actually, come to think of it we could probably future-proof this command against this kind of thing by just always using the `dartSdkVersion` as this version.

Reply:
I realized that wouldn't be quite right either; it would break us in cases like when Dart supported NNBD, but the template hadn't been updated yet and was still using an older version constraints so as not to opt in. https://github.com/flutter/plugins/pull/5281 should be the fully future-proof version.